frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx

```diff
@@ -98,20 +136,44 @@ export default function CourseDescription({ structure }) {
         prepareUserPayload(degree, planner)
       );
       console.log(data);
-      if (!err) setCoursesPathTo({
-        "direct_unlock": data.direct_unlock,
-        "indirect_unlock": data.indirect_unlock
-      });
+      if (!err)
+        setCoursesPathTo({
+          direct_unlock: data.direct_unlock,
+          indirect_unlock: data.indirect_unlock,
+        });
+    };
+
+    const getCourseCapacityById = async (id) => {
+      const [data, err] = await axiosRequest(
+        "get",
+        `http://localhost:3001/api/terms/2022-T2/courses/${id}`
+      );
+      console.log(data);
+      if (!err) {
+        getCapacityAndEnrolment(data);
+      }
     };
 
-    setpageLoaded(false);
     if (id === "explore" || id === "search") return;
     if (id) {
       getCourse();
       getPathToCoursesById(id);
+      getCourseCapacityById(id);
     }
+    setPageLoaded(true);
   }, [id]);
 
+  const getCapacityAndEnrolment = (data) => {
+    let enrolmentCapacityData = {};
+    for (let i = 0; i < data.classes.length; i++) {
+      if (data.classes[i].activity === "Lecture") {
+        enrolmentCapacityData = data.classes[i].courseEnrolment;
```

Comment:
this is an ok workaround - but you should notice this will fail if a course doesnt run lectures. Also, this assumes that there is only 1 lecture. You should come up with a way to deal with both those cases.